src/platform/site-wide/user-nav/components/SearchMenu.jsx

```diff
@@ -73,6 +76,19 @@ export class SearchMenu extends React.Component {
     this.setState({ userInput: event.target.value });
   };
 
+  handleKeyUp = event => {
+    if (
+      (event.which || event.keyCode) === ENTER_KEY &&
+      document.getElementById('query') === document.activeElement
+    ) {
+      this.handleSearchEvent();
+      return;
+    }
+    if ((event.which || event.keyCode) === ESCAPE_KEY) {
+      this.props.clickHandler();
+    }
+  };
+
```

Comment:
Instead of leveraging keyUp I was able to get the onSubmit handler working by removing the `onSubmit` from the button and adding it into the `<form` like 

```
// line 123
if (!this.props.searchTypeaheadEnabled) {
      return (
        <form acceptCharset="UTF-8" onSubmit={(event) => { this.handleSearchEvent(); event.preventDefault(); }}>
          <label htmlFor="query" className="usa-sr-only">
            Search:
          </label>
// ...
```

I think you maybe you needed the `event.preventDefault();`?

Reply:
Oh gosh, I just reread my logic and realized this is when typeahead is disabled 🤦

Reply:
no worries at all! that form is still there for when we want the new experience disabled, but eventually it will go away! I am happy to make changes to it however